### PR TITLE
Rename from strategies to intents

### PR DESCRIPTION
**Pull request by MatiasOS**

### CODE REVIEW

File: src/Types.ts [hunk 1]
```diff
@@ -241,64 +241,64 @@ export type IntentArgs = {
...
+export type SegmentArgs = {
+  functionName: SegmentFunctionName
+  params: Record<string, SegmentParamValue>
   data?: string
   requiresUnsignedCall?: boolean
 }
 
-export type OrderArgs = {
-  primitives: PrimitiveArgs[]
+export type DeclarationIntentArgs = {
```
Comment (mikec) [inline src/Types.ts:251]: anything that was "order" should be "intent" so this should be `IntentArgs`

File: src/Types.ts [hunk 2]
```diff
@@ -241,64 +241,64 @@ export type IntentArgs = {
...
+export type SegmentJSON = {
+  functionName: SegmentFunctionName
+  params: Record<string, SegmentParamJSON>
   data: string
   requiresUnsignedCall: boolean
 }
 
-export type OrderJSON = {
-  primitives: PrimitiveJSON[]
+export type DeclarationIntentJSON = {
```
Comment (mikec) [inline src/Types.ts:281]: should be `IntentJSON`

**Review by mikec — CHANGES_REQUESTED** [review]
small change then this is ready to go!

File: src/Types.ts [hunk 3]
```diff
@@ -241,64 +241,64 @@ export type IntentArgs = {
   expiryBlock?: BigIntish
```
Comment (mikec) [inline src/Types.ts:241]: the DSL args can be named differently so they don't conflict. Also it looks like they're named at the wrong levels:

`IntentArgs` would be:

```
export type DeclarationDefinitionArgs = {
  intents: IntentSegmentArgs[]
  replay?: IntentReplay
  expiryBlock?: BigIntish
}

and `IntentSegmentArgs` would be:

```
export type IntentDefinitionArgs = {
  replay?: IntentReplay
  expiryBlock?: BigIntish
  conditions?: ConditionArgs[]
  actions: ActionArgs[]
}
```

**Review by phanpak — APPROVED** [review]

**Review by mikec — APPROVED** [review]

**Pull request merged**